src/main/java/com/adobe/aem/compgenerator/utils/CommonUtils.java

```diff
@@ -251,4 +255,29 @@ private static BufferedWriter getFileWriterFromTemplate(String path,
         }
         return null;
     }
+    
+    
+    /**
+     * Gets the Tab properties if defined else returns all the properties.
+     *
+     * @param properties The {@link Property}
+     * @param tabs The {@link Tab}
+     * @return List<Property>
+     */
+    public static List<Property> getTabbedPropertiesIfExists(List<Property> properties, List<Tab> tabs) {
+        List<Property> updatedPropertiesList = null;
+        if (null != tabs && !tabs.isEmpty()) {
+            List<Property> sortedProperties = new ArrayList<>();
+            for (Tab tab : tabs) {
+                tab.getFields().stream()
+                        .filter(field -> properties.stream().anyMatch(
+                                property -> property.getField().equals(field) ? sortedProperties.add(property) : false))
+                        .collect(Collectors.toList());
```

Comment:
Similar to a comment I had elsewhere, no need to `collect()` if we're not returning the result into a variable assignment.

We can either remove the `.collect()` or simplify the streaming code to return a value that we add to `sortedProperties`

Reply:
Updated the logic to sort it differently instead of collecting again to a different list